src/platform/macos/window.rs

```diff
@@ -861,55 +890,38 @@ impl Window2 {
     }
 
     #[inline]
-    pub fn get_inner_size(&self) -> Option<(u32, u32)> {
-        let factor = self.hidpi_factor() as f64; // API convention is that size is in physical pixels
-        unsafe {
-            let view_frame = NSView::frame(*self.view);
-            Some(((view_frame.size.width*factor) as u32, (view_frame.size.height*factor) as u32))
-        }
+    pub fn get_inner_size(&self) -> Option<LogicalSize> {
+        let view_frame = unsafe { NSView::frame(*self.view) };
+        Some((view_frame.size.width as f64, view_frame.size.height as f64).into())
     }
 
     #[inline]
-    pub fn get_outer_size(&self) -> Option<(u32, u32)> {
-        let factor = self.hidpi_factor() as f64; // API convention is that size is in physical pixels
-        unsafe {
-            let window_frame = NSWindow::frame(*self.window);
-            Some(((window_frame.size.width*factor) as u32, (window_frame.size.height*factor) as u32))
-        }
+    pub fn get_outer_size(&self) -> Option<LogicalSize> {
+        let view_frame = unsafe { NSWindow::frame(*self.window) };
+        Some((view_frame.size.width as f64, view_frame.size.height as f64).into())
     }
 
     #[inline]
-    pub fn set_inner_size(&self, width: u32, height: u32) {
-        let factor = self.hidpi_factor() as f64; // API convention is that size is in physical pixels
+    pub fn set_inner_size(&self, size: LogicalSize) {
         unsafe {
-            NSWindow::setContentSize_(*self.window, NSSize::new((width as f64)/factor, (height as f64)/factor));
+            NSWindow::setContentSize_(*self.window, NSSize::new(size.width as CGFloat, size.height as CGFloat));
         }
     }
 
-    pub fn set_min_dimensions(&self, dimensions: Option<(u32, u32)>) {
+    pub fn set_min_dimensions(&self, dimensions: Option<LogicalSize>) {
         unsafe {
-            let (width, height) = dimensions.unwrap_or((0, 0));
-            nswindow_set_min_dimensions(self.window.0, width.into(), height.into());
+            let dimensions = dimensions.unwrap_or_else(|| (!0, !0).into());
```

Comment:
Hello, I think this line should be:
```rust
let dimensions = dimensions.unwrap_or_else(|| (0, 0).into());
```

Otherwise `window.set_min_dimensions(None)` [crashes the application](https://gist.github.com/yvt/e54982a9199c95fb4678461a66e48edf) by changing the window size to a huge value.

Reply:
Whoops, good catch. Looks like I got carried away with copy-pasting.